cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/services/http/handlers/WorkflowStatsSlaHttpHandlerTest.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.internal.app.services.http.handlers;
+
+import co.cask.cdap.WorkflowApp;
+import co.cask.cdap.app.store.Store;
+import co.cask.cdap.common.app.RunIds;
+import co.cask.cdap.common.conf.Constants;
+import co.cask.cdap.internal.app.services.http.AppFabricTestBase;
+import co.cask.cdap.internal.app.store.DefaultStore;
+import co.cask.cdap.internal.app.store.WorkflowDataset;
+import co.cask.cdap.proto.Id;
+import co.cask.cdap.proto.ProgramRunStatus;
+import co.cask.cdap.proto.ProgramType;
+import com.google.gson.reflect.TypeToken;
+import org.apache.http.HttpResponse;
+import org.apache.twill.api.RunId;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Tests for {@link co.cask.cdap.gateway.handlers.WorkflowStatsSLAHttpHandler}
+ */
+public class WorkflowStatsSLAHttpHandlerTest extends AppFabricTestBase {
+
+  @Test
+  public void testStatistics() throws Exception {
+
+    Store store = getInjector().getInstance(DefaultStore.class);
+
+    deploy(WorkflowApp.class);
+    String workflowName = "FunWorkflow";
+    String mapreduceName = "ClassicWordCount";
+    String sparkName = "SparkWorkflowTest";
+
+    Id.Program workflowProgram =
+      Id.Workflow.from(Id.Namespace.DEFAULT, "WorkflowApp", ProgramType.WORKFLOW, workflowName);
+    Id.Program mapreduceProgram =
+      Id.Program.from(Id.Namespace.DEFAULT, "WorkflowApp", ProgramType.MAPREDUCE, mapreduceName);
+    Id.Program sparkProgram =
+      Id.Program.from(Id.Namespace.DEFAULT, "WorkflowApp", ProgramType.SPARK, sparkName);
+
+    for (int i = 0; i < 10; i++) {
+      RunId workflowRunId = RunIds.generate();
+      store.setStart(workflowProgram, workflowRunId.getId(), RunIds.getTime(workflowRunId, TimeUnit.SECONDS));
```

Comment:
If you're testing Http Handler code using AppFabricTestBase, you can just call the REST APIs to start workflows. Any specific reason for using the store directly like this? the start workflow API will do all these store updates for you...